### DISCUSSION TITLE
Decoupling form validation and error messages

### DISCUSSION THREAD

**Discussion by dmgauthier** [mantinedev/mantine#2583] [Feature requests]
Hi there !! 

We recently decided to replace our own design system ( that we maintained for years ) with Mantine, and I must say, its great ! 

I wanna discuss something about the the validation of a form. Specifically about decoupling the error messages rendering and the validation state of the form per say.

e.g.
A form has its save button enabled only when the form is valid. In this case, we better make sure the form validation is evaluated onChange. But by doing so : if a numerical input is valid only between `1000` and `2000`, an error message will appear as soon as a user writes `1` ; if a text input only takes URLs the error message will show as soon as the user enters `h` . 

You understand that this is very aggressive UX wise. So a very nice plus IMO would be to have the possiblity to show validation messages onBlur, while still evaluating the form validation onChange.

Thanks again for this awesome lib, its a bless to work with.

**Comment by rtivital** [comment]
Use `form.isValid` handler to get current validation state without showing errors – https://mantine.dev/form/validation/#isvalid-handler
